modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -3353,9 +3353,14 @@ else if (tx.setRollbackOnly())
          * @return True if message required, false otherwise.
          */
         private boolean isMvccRecoveryMessageRequired() {
-            return node.isClient() && mvccCrd != null && mvccCrd.nodeId() != null &&
+            ClusterNode mvccCrdNode = null;
+
+            if (mvccCrd != null && mvccCrd.nodeId() != null)
+                mvccCrdNode = cctx.node(mvccCrd.nodeId());
+
+            return node.isClient() && mvccCrdNode != null &&
                 (cctx.kernalContext().coordinators().mvccEnabled() ||
-                    !IgniteFeatures.nodeSupports(cctx.node(mvccCrd.nodeId()), IgniteFeatures.MVCC_TX_RECOVERY_PROTOCOL_V2));
+                    !IgniteFeatures.nodeSupports(mvccCrdNode, IgniteFeatures.MVCC_TX_RECOVERY_PROTOCOL_V2));
```

Comment:
Please remind me why do we need this at all? Apache Ignite does not have compatibility between versions.

Can we get rid of feature check entirely, assume it as "always true"?

Reply:
I am not sure that it is related to comatibility. It was introduced as an optimization within fixes for https://issues.apache.org/jira/browse/IGNITE-12350

Reply:
Please try to remove IgniteFeatures use, see if nothing is broken.

Reply:
Using IgniteFeatures was removed. It looks like nothing is broken

Reply:
Also, I changed the name of the test method